specification/peering/resource-manager/Microsoft.Peering/preview/2019-03-01-preview/peering.json

```diff
@@ -0,0 +1,1309 @@
+        "operationId": "GetPeerInfo",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/PeerInfo"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get peer info": {
+            "$ref": "./examples/GetPeerInfo.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/updatePeerInfo": {
+      "post": {
+        "tags": [
+          "PeerInfo"
+        ],
+        "description": "Updates the peer info associated with the specified subscription.",
+        "operationId": "UpdatePeerInfo",
+        "parameters": [
+          {
+            "name": "peerInfo",
+            "in": "body",
+            "description": "The peer info.",
+            "required": true,
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/PeerInfo"
+              }
+            }
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/PeerInfo"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation has failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update peer info": {
+            "$ref": "./examples/UpdatePeerInfo.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Peering/peeringLocations": {
+      "get": {
+        "tags": [
+          "PeeringLocations"
+        ],
+        "description": "Lists all of the available peering locations for the specified kind of peering.",
+        "operationId": "PeeringLocations_List",
+        "parameters": [
+          {
+            "name": "kind",
+            "in": "query",
+            "description": "The kind of the peering.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "Direct",
+              "Exchange"
+            ]
```

Comment:
is there a schema difference of the resource , based on the Kind ? 
If yes, then ensure to define the peeringLocation resource schema using 'discriminator' pattern.. Search for x-ms-discriminator-value in the repo.

Reply:
based on the kind direct attributes will be null. Direct Kind means the PeeringDirectProperties cannot be null and exchange must be null.

Reply:
@sanjaiganesh does this work for you?

Reply:
It is not ideal. Please check discriminator pattern that I suggested. It will avoid having this additional layer (  "Direct" :  {}  and "Exchange" :  {})


---
In reply to: [266659045](https://github.com/Azure/azure-rest-api-specs/pull/5343#discussion_r266659045) [](ancestors = 266659045)